sphinx/tutorial/rose/furthertopics/widget-development.rst

```diff
@@ -0,0 +1,156 @@
+Widget Development
+==================
+
+
+The ``rose edit`` GUI displays configurations using built-in widgets. For more
+complex requirements ``rose edit`` supports custom widgets as plugins.
+
+In this tutorial we will write a custom widget which offers typing suggestions
+when entering usernames.
+
+.. image:: img/rose-edit-custom-widget.png
+   :align: center
+   :width: 450px
+
+.. warning::
+
+   If you find yourself needing to write a custom widget, please contact the
+   Rose team for guidance.
+
+Example
+-------
+
+Create a new rose app by running the following command replacing
+``DIRECTORY`` with the path in which to create the suite::
+
+   rose tutorial widget DIRECTORY
+   cd DIRECTORY
+
+You will now have a rose app which contains the following files:
+
+.. code-block:: none
+
+   DIRECTORY
+   |-- meta
+   |   `-- lib
+   |       `-- python
+   |           `-- widget
+   |               |-- __init__.py
+   |               `-- username.py
+   `-- rose-app.conf
+
+The ``rose-app.conf`` file defines an environment variable called ``USER``:
+
+.. code-block:: rose
+
+   [env]
+   USER=fred
+
+.. _python package: https://docs.python.org/3/tutorial/modules.html#packages
+
+The ``__init__.py`` file is empty - the presence of this file declares the
+``widget`` directory as a `python package`_.
+
+The ``username.py`` file is where we will write our widget.
+
+Initial Code
+^^^^^^^^^^^^
+
+We will start with a slimmed-down copy of the class
+``rose.config_editor.valuewidget.text.RawValueWidget`` which you will find
+in the file ``username.py``. It contains all the API calls you would normally
```

Comment:
As mentioned in a previous comment, 'API' is unnecessarily technical in my opinion.